src/libraries/System.Text.Json/Common/JsonSourceGenerationOptionsAttribute.cs

```diff
@@ -43,11 +43,16 @@ class JsonSerializerOptionsAttribute : JsonAttribute
         /// <summary>
         /// Specifies a built-in naming polices to convert JSON property names with.
         /// </summary>
-        public JsonKnownNamingPolicy NamingPolicy { get; set; }
+        public JsonKnownNamingPolicy PropertyNamingPolicy { get; set; }
 
         /// <summary>
         /// Specifies whether JSON output should be pretty-printed.
         /// </summary>
         public bool WriteIndented { get; set; }
+
+        /// <summary>
+        /// Specifies the source generation mode.
+        /// </summary>
+        public JsonSourceGenerationMode GenerationMode { get; set; } = JsonSourceGenerationMode.Metadata | JsonSourceGenerationMode.Serialization;
```

Comment:
This change addresses my thoughts from https://github.com/dotnet/aspnetcore/pull/32374#discussion_r646636567. This new API makes it cleaner & easier to test https://github.com/dotnet/runtime/pull/54526.